Dockerfile

```
@@ -13,105 +13,148 @@
 # limitations under the License.
 
 ###############################################################################
-# Stage 1: Create the develop, test, and build environment
+# Stage 1: Create the developer image for the BUILDPLATFORM only
 ###############################################################################
-FROM  registry.access.redhat.com/ubi8/ubi-minimal:8.4 AS develop
+ARG GOLANG_VERSION=1.17
+FROM --platform=$BUILDPLATFORM registry.access.redhat.com/ubi8/go-toolset:$GOLANG_VERSION AS develop
 
-ARG GOLANG_VERSION=1.17.13
 ARG PROTOC_VERSION=21.5
 
 USER root
 
 # Install build and dev tools
-RUN microdnf install \
-    gcc \
-    gcc-c++ \
-    make \
-    vim \
-    findutils \
-    diffutils \
-    git \
-    wget \
-    tar \
-    unzip \
-    python3 \
-    nodejs && \
-    pip3 install pre-commit
-
-# Install go
-ENV PATH /usr/local/go/bin:$PATH
-RUN set -eux; \
-    wget -qO go.tgz "https://golang.org/dl/go${GOLANG_VERSION}.linux-amd64.tar.gz"; \
-    sha256sum *go.tgz; \
-    tar -C /usr/local -xzf go.tgz; \
-    go version
+RUN --mount=type=cache,target=/root/.cache/dnf:rw \
+    dnf install -y --nodocs --setopt=cachedir=/root/.cache/dnf \
+        python3 \
+        python3-pip \
+        nodejs \
+    && true
+
+# Install pre-commit
+ARG PIP_CACHE_DIR=/root/.cache/pip
+RUN --mount=type=cache,target=$PIP_CACHE_DIR \
+    pip3 install pre-commit && \
+    pip3 list
```

NIT: assuming `pip3 list` was put here for debugging and should be removed?
```suggestion
    pip3 install pre-commit
```

---

Gone.